packages/amplify-graphql-auth-transformer/src/utils/index.ts

```diff
@@ -47,9 +47,11 @@ export const getAuthDirectiveRules = (authDir: DirectiveWrapper, options?: GetAu
       operations.splice(indexOfRead, 1);
       operations.push('get');
       operations.push('list');
-      operations.push('search');
-      operations.push('sync');
       operations.push('listen');
+      if (!options.isSqlDataSource) {
```

Comment:
Since `options` is an optional input, this will fail if it's not passed. Would recommend `if (!options?.isSqlDataSource) {`

Reply:
Thanks for catching that! I wonder why the ACM tests are passing, since one of the `getAuthDirectiveRules` invocations inside of `printAcm` doesn't pass args. (And for that matter, I wonder why `tsc` didn't complain about this?)